src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -264,6 +269,9 @@ private void Ldc(int i)
             }
         }
 
+        /// <summary>A macro for _ilg.Emit(OpCodes.Ldc_I8).</summary>
+        private void LdcI8(long i) => _ilg!.Emit(OpCodes.Ldc_I8, i);
```

Comment:
aside, why not change `_ilg` to be set via a protected property or method, so it could be null checked once and you could remove all these bangs.

Reply:
Are you suggesting that the getter would do a null check?  If so, it would check it every access.

Or are you suggesting that the setter do a null check and the getter be declared to be non-nullable?  If so, that wouldn't help, as there's no guarantee the setter would have been called.

The issue here is the code expects the derived class to have set the protected field, which starts life as null.  We could of course lie and say that the field isn't nullable, but then, what's the point of the nullable reference types.